File: backend/credit-service/src/main/java/com/bank/credit/dto/CreateCreditRequest.java
package com.bank.credit.dto;

import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;

import java.math.BigDecimal;

public record CreateCreditRequest(
        @NotNull(message = "ID пользователя обязателен")
        Long userId,

        @NotNull(message = "ID счёта обязателен")
        Long accountId,

        @NotNull(message = "ID тарифа обязателен")
        Long tariffId,

        @NotNull(message = "Сумма кредита обязательна")
        @DecimalMin(value = "1", message = "Сумма должна быть положительной")
        BigDecimal amount,

        @Min(value = 1, message = "Срок должен быть минимум 1 день")
        int termDays
) {}

File: ios/BankShared/Sources/BankShared/DTOs/Requests/CreateCreditRequest.swift
import Foundation

public struct CreateCreditRequest: Codable, Sendable {
    public let accountId: Int64
    public let tariffId: Int64
    public let amount: Decimal
    public let termDays: Int

    public init(accountId: Int64, tariffId: Int64, amount: Decimal, termDays: Int) {
        self.accountId = accountId
        self.tariffId = tariffId
        self.amount = amount
        self.termDays = termDays
    }
}
